src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.h

```diff
@@ -0,0 +1,688 @@
+     *
+     *   @param activationReason [in]       Provide information as to why the transport was started or stopped.
+     *
+     *   @param timeControl   [in]          Configuration to control the life cycle of a triggered transport.
+     *
+     *   The delegate is expected to begin transmission using the timeControl values.
+     *
+     *   Emitting of PushTransportBegin event is handled by the server when delegate returns success.
+     *
+     *   The delegate should emit PushTransportEnd Event using GeneratePushTransportEndEvent() API when timeControl values when
+         transmission ends.
+     *   @return Success if the stream transport trigger is successful; otherwise, the command SHALL be rejected with Failure.
+     */
+    virtual Protocols::InteractionModel::Status
+    ManuallyTriggerTransport(const uint16_t connectionID, TriggerActivationReasonEnum activationReason,
+                             const Optional<Structs::TransportMotionTriggerTimeControlStruct::Type> & timeControl) = 0;
+
+    /**
+     * @brief Validates the url
+     * @param[in] url The url to validate
+     *
+     * @return boolean value true if the url is valid else false.
+     */
+    virtual bool ValidateUrl(std::string url) = 0;
+
+    /**
+     * @brief Validates the bandwidth requirement against the camera's resource management
+     *
+     * The implementation SHALL ensure:
+     *  - The requested stream usage (streamUsage) is allowed given the current allocation of
+     *    camera resources (e.g. CPU, memory, network bandwidth).
+     *
+     * @param[in] streamUsage    The desired usage type for the stream (e.g. live view, recording, etc.).
+     * @param[in] videoStreamId  Optional identifier for the requested video stream.
+     * @param[in] audioStreamId  Optional identifier for the requested audio stream.
+     *
+     * @return Status::Success if the stream usage is valid; and Status::ResourceExhausted otherwise.
+     */
+
+    virtual Protocols::InteractionModel::Status
+    ValidateBandwidthLimit(StreamUsageEnum streamUsage, const Optional<DataModel::Nullable<uint16_t>> & videoStreamId,
+                           const Optional<DataModel::Nullable<uint16_t>> & audioStreamId) = 0;
+
+    /**
+     * @brief Assign an existing Video Stream as per the camera's resource management and stream priority policies using requested
+     * stream usage.
+     *
+     * @param[in] streamUsage    The desired usage type for the stream (e.g. live view, recording, etc.).
+     * @param[in] videoStreamId  Identifier for the requested video stream.
+     *
+     * @return Assign the selected videoStreamID and return Status::Success. Return Status::InvalidStream if there are no allocated
+     * VideoStream.
+     */
+    virtual Protocols::InteractionModel::Status SelectVideoStream(StreamUsageEnum streamUsage, uint16_t & videoStreamId) = 0;
+    /**
```

Comment:
The spacing around these various comments is odd.  I would expect blank line above comment, no blank line between comment and the function it's documenting.